src/mono/mono/mini/aot-compiler.c

```diff
@@ -15122,17 +15112,7 @@ aot_assembly (MonoAssembly *ass, guint32 jit_opts, MonoAotOptions *aot_options)
 		acfg->flags = (MonoAotFileFlags)(acfg->flags | MONO_AOT_FILE_FLAG_LLVM_ONLY);
 
 	acfg->assembly_name_sym = g_strdup (get_assembly_prefix (acfg->image));
-#ifdef TARGET_WASM
-	acfg->global_prefix = g_strdup_printf ("mono_aot_%s", g_strdup(mono_fixup_symbol_name (acfg->assembly_name_sym)));
-#else
-	char *p;
-    /* Get rid of characters which cannot occur in symbols */
-	for (p = acfg->assembly_name_sym; *p; ++p) {
-		if (!(isalnum (*p) || *p == '_'))
-			*p = '_';
-	}
-	acfg->global_prefix = g_strdup_printf ("mono_aot_%s", acfg->assembly_name_sym);
-#endif
+	acfg->global_prefix = mono_fixup_symbol_name ("mono_aot_", acfg->assembly_name_sym, "");
```

Comment:
@lambdageek I'm not sure if here also is need to free the memory,  looks like `emit_aot_image` will take care of it ?

Reply:
@mkhamoyan looks like `acfg_free()` needs to be updated to free `acfg->global_prefix`